### PR TITLE
EOS:18596 - Cleanup logs in Reset phase in s3setup

### PR DESCRIPTION
**Pull request by vimalraghwani20**
s3 -> /var/log/seagate/s3/
Background -> /var/log/seagate/s3/s3backgrounddelete/
Auth -> /var/log/seagate/auth/
Slapd -> /var/log/slapd.log
HAproxy -> /var/log/haproxy.log
Audit -> /var/log/audit/
Crash dumps -> /var/log/crash/core-s3server.*.gz
Motr -> /var/log/seagate/motr/

Test O/P:

**[root@ssc-vm-1940 seagate]# /opt/seagate/cortx/s3/bin/s3_setup reset --config "json:///root/s3confstore.json"**
Running validations..
INFO: post_install validations successful.
Processing reset json:///root/s3confstore.json
INFO: Cleaning up log files.
INFO:Log files cleanup successful.
INFO: reset successful.
**[root@ssc-vm-1940 seagate]# /opt/seagate/cortx/s3/bin/s3_setup cleanup --config "json:///root/s3confstore.json"**
Running validations..
INFO: post_install validations successful.
Processing cleanup json:///root/s3confstore.json
LDAP account: s3-background-delete-svc does not exist, skipping deletion.
INFO: Reverting config files.
INFO: Reverting config files successful.
INFO: cleanup successful.
[root@ssc-vm-1940 seagate]#

### CODE REVIEW

File: scripts/provisioning/resetcmd.py [hunk 1]
```diff
@@ -35,6 +38,64 @@ def __init__(self, config: str):
...
+    # s3 -> /var/log/seagate/s3/
+    if os.path.exists("/var/log/seagate/s3"):
+      self.DeleteDir("/var/log/seagate/s3/")
+
+    #Auth -> /var/log/seagate/auth/
+    if os.path.exists("/var/log/seagate/auth"):
+      self.DeleteDir("/var/log/seagate/auth/")
+
+    #S3 Motr -> /var/log/seagate/motr/
+    if os.path.exists("/var/log/seagate/motr"):
```
Comment (vimalraghwani20) [inline scripts/provisioning/resetcmd.py:70]: remove only s3server-* in the motr dir

File: scripts/provisioning/resetcmd.py [hunk 2]
```diff
@@ -35,6 +38,64 @@ def __init__(self, config: str):
...
+      sys.stdout.write('INFO: Cleaning up log files.\n')
+      self.CleanupLogs()
+      sys.stdout.write('INFO:Log files cleanup successfull.\n')
+    except Exception as e:
+      sys.stderr.write(f'Failed to cleanup log directories or files, error: {e}\n')
+      raise e
+
+
+  def CleanupLogs(self):
+    """Cleanup all the log directories and files."""
```
Comment (nileshgovande) [inline scripts/provisioning/resetcmd.py:52]: Use loop aver array

File: scripts/provisioning/resetcmd.py [hunk 3]
```diff
@@ -35,6 +38,64 @@ def __init__(self, config: str):
...
+    if os.path.exists("/var/log/slapd.log"):
+      os.remove("/var/log/slapd.log")
+
+    #S3 Crash dumps -> /var/log/crash/core-s3server.*.gz
+    if os.path.exists("/var/log/crash"):
+      files = glob.glob('/var/log/crash/core-s3server.*.gz')
+      for file in files:
+        os.remove(file)
+
+  def DeleteDir(self, dirname: str):
```
Comment (nileshgovande) [inline scripts/provisioning/resetcmd.py:90]: DeleteDirContents

Comment (vimalraghwani20) [reply scripts/provisioning/resetcmd.py]: fixed.

File: scripts/provisioning/resetcmd.py (hunk 1 above)
Comment (vimalraghwani20) [reply scripts/provisioning/resetcmd.py]: fixed.

File: scripts/provisioning/resetcmd.py [hunk 4]
```diff
@@ -35,6 +38,84 @@ def __init__(self, config: str):
 
   def process(self):
     """Main processing function."""
-    retval = 0
     sys.stdout.write(f"Processing {self.name} {self.url}\n")
-    return retval
+    try:
+      sys.stdout.write('INFO: Cleaning up log files.\n')
+      self.CleanupLogs()
+      sys.stdout.write('INFO:Log files cleanup successfull.\n')
```
Comment (amitwac2608) [inline scripts/provisioning/resetcmd.py:45]: successful

Comment (amitwac2608) [reply scripts/provisioning/resetcmd.py:45]: Also fix codacy issues.

File: scripts/provisioning/resetcmd.py (hunk 2 above)
Comment (vimalraghwani20) [reply scripts/provisioning/resetcmd.py:52]: fixed.

File: scripts/provisioning/resetcmd.py [hunk 5]
```diff
@@ -35,6 +38,84 @@ def __init__(self, config: str):
...
+    # s3 -> /var/log/seagate/s3/
+    #Auth -> /var/log/seagate/auth/
+    #S3 Motr -> /var/log/seagate/motr/s3server-*
+    #HAproxy -> /var/log/haproxy.log
+    #HAproxy -> /var/log/haproxy-status.log
+    #Slapd -> /var/log/slapd.log
+    #S3 Crash dumps -> /var/log/crash/core-s3server.*.gz
+
+    logFolders = ["/var/log/seagate/s3/s3backgrounddelete",
+                  "/var/log/seagate/s3/audit",
```
Comment (nileshgovande) [inline scripts/provisioning/resetcmd.py:64]: ????

Comment (vimalraghwani20) [reply scripts/provisioning/resetcmd.py]: fixed redundant code

**Review by sachitanands — COMMENTED** [review]
looks good

**Pull request merged**